### PR TITLE
Bugfix: use correct offset to extract bundle ID

### PR DESCRIPTION
**Pull request by weiqiangt**

### CODE REVIEW

**Comment by weiqiangt** [comment]
@mengdie-song @tnqn

**Review by mengdie-song — COMMENTED** [review]
LGTM

File: ofctrl/ofSwitch.go [hunk 1]
```diff
@@ -357,6 +357,8 @@ func (self *OFSwitch) handleMessages(dpid net.HardwareAddr, msg util.Message) {
 				log.Errorf("Received Vendor error: %s on ONFT_BUNDLE_ADD_MESSAGE message", errMsg)
 			}
 		default:
+			result.msgType = UnknownMessage
+			self.publishMessage(0, result)
```
Comment (tnqn) [inline ofctrl/ofSwitch.go]: @weiqiangt where will the message be received since it doesn't have a valid bundleID?

Comment (weiqiangt) [reply ofctrl/ofSwitch.go]: Sorry, I used to wrongly assumed the xID is different from the bundle ID.

File: ofctrl/transaction.go [hunk 2]
```diff
@@ -121,23 +121,29 @@ func (tx *Transaction) listenReply() {
...
-				select {
-				case tx.controlIntCh <- reply:
-				case <-time.After(messageTimeout):
-					log.Warningln("BundleControlMessage reply message accept timeout")
+			if func() bool {
+				tx.lock.Lock()
+				defer tx.lock.Unlock()
+
+				if !ok { // controlReplyCh closed.
+					return true
```
Comment (tnqn) [inline ofctrl/transaction.go]: Does it need to hold the lock?

File: ofctrl/transaction.go [hunk 3]
```diff
@@ -121,23 +121,29 @@ func (tx *Transaction) listenReply() {
...
-			}
-			switch reply.msgType {
-			case BundleControlMessage:
-				select {
-				case tx.controlIntCh <- reply:
-				case <-time.After(messageTimeout):
-					log.Warningln("BundleControlMessage reply message accept timeout")
+			if func() bool {
+				tx.lock.Lock()
+				defer tx.lock.Unlock()
```
Comment (tnqn) [inline ofctrl/transaction.go]: Normally it's not good to hold the lock when there is a sleep. Could you add comments to explain what the lock protects from?

File: ofctrl/transaction.go (hunk 2 above)
Comment (weiqiangt) [reply ofctrl/transaction.go]: Just found we don't need to lock that many, updated the lock scope.

**Review by tnqn — APPROVED** [review]
LGTM

**Review by mengdie-song — APPROVED** [review]

**Pull request merged**